x-pack/plugins/alerting/server/rules_client/rules_client.ts

```diff
@@ -2035,6 +2035,23 @@ export class RulesClient {
       } catch (e) {
         throw e;
       }
+    }
+
+    let scheduledTaskIdToCreate: string | null = null;
+    if (attributes.scheduledTaskId) {
+      // If scheduledTaskId defined in rule SO, make sure it exists
+      try {
+        await this.taskManager.get(attributes.scheduledTaskId);
```

Comment:
@pmuellr Good point. Do you think this extra check is worth doing? I think the possibility that a scheduled task id is defined but the document doesn't exist should be small (I think the user would have to be messing around with the task manager index), so this was just to cover all scenarios, but maybe not needed?

Reply:
Possibility does seem low, but if for some reason it doesn't exist, it would otherwise never get created, right?  So I think we still need to do this.

Reply:
Created issue for checking if task doc exists: https://github.com/elastic/kibana/issues/140505